pennylane/interfaces/execution.py

```diff
@@ -740,10 +762,24 @@ def device_gradient_fn(inner_tapes, **gradient_kwargs):
         # within execute_and_gradients, so providing a gradient_fn
         # in this case would have ambiguous behaviour.
         raise ValueError("Gradient transforms cannot be used with grad_on_execution=True")
+    elif interface in jpc_interfaces:
+        cache_full_jacobian = interface == "autograd"
+        jpc = TransformJacobianProducts(
+            execute_fn, gradient_fn, gradient_kwargs, cache_full_jacobian=cache_full_jacobian
+        )
+        for _ in range(1, max_diff):
+            ml_boundary_execute = _get_ml_boundary_execute(interface, _grad_on_execution)
+            execute_fn = partial(ml_boundary_execute, execute_fn=execute_fn, jpc=jpc)
+            jpc = TransformJacobianProducts(execute_fn, gradient_fn, gradient_kwargs)
+            cache_full_jacobian = False

```

Comment:
this whole looping section feels strange to me. we're nesting `jpc` instances within `jpc` instances to achieve higher-order derivatives? I think a comment would help here, but I'm also wondering if there's a better way to encapsulate higher-order derivative logic in the `TransformJacobianProducts` class instead of `qml.execute`. thoughts?

Reply:
I'm only caching the jacobian on the first iteration due to this line: 

https://github.com/PennyLaneAI/pennylane/blob/07174730ec059d0a2c7264922ede773923a7a5d7/pennylane/interfaces/autograd.py#L228

Don't know why, but this seems like a behavior I shouldn't mess with.

Reply:
shouldn't it only be on the last loop iteration then, not the first? also, do you know why that original code is specific to parameter shift only?